onnxruntime/core/framework/data_types.cc

```diff
@@ -685,33 +685,33 @@ const char* DataTypeImpl::ToString(MLDataType type) {
   if (prim_type != nullptr) {
     switch (prim_type->GetDataType()) {
       case TensorProto_DataType_FLOAT:
-        return "tensor(float)";
+        return "float";
```

Comment:
seems weird to include "tensor" when it is supposed to just return the string associated with the primitive type alone. Seems right to fix this inaccuracy.

Reply:
It's been awhile since I looked at this, but I believe this was because in ONNX, data types for inputs/outputs/initializers must be tensors. (for ML, there were also maps, sequences etc.) , they couldn't be primitive types.  
https://github.com/onnx/onnx/blob/master/docs/IR.md#standard-data-types

Reply:
is it trying to be consistent with https://github.com/onnx/onnx/blob/master/onnx/defs/data_type_utils.cc#L121 ?

Reply:
Thanks for the info. 
In the above method I believe there is switching on the high level type again - tensors, sequences, and maps. Here, the switch is only the primitive type and I believe including "tensor" arbitrarily is not right. If it is not a primitive type, it will try to create the string from the ONNX TypeProto.

Reply:
The method you are referring to above is being invoked IF the given type is not a primitive type. So, I think "tensor" being added while switching on the primitive type is quite unnecessary.

Reply:
thanks. that makes sense. I guess AsPrimitiveType() returns non-null for primitive types.

Reply:
yes